src/ia-anniversary-banner.ts

```diff
@@ -0,0 +1,197 @@
+
+import './marquee';
+
+@customElement('ia-anniversary-banner')
+export class IaAnniversaryBanner extends LitElement {
+  @property({ type: Object }) iaFiles: IAFile[] = [];
+
+  @property({ type: Object }) iaMD: IAMD = { directory: '', separator_dir: '' };
+
+  @property({ type: Object }) moments: { [key: string]: TimelineMoment } = {};
+
+  @property({ type: Array }) shuffledMoments: TimelineMoment[] = [];
+
+  @property({ type: String }) landingURL = 'https://archive.org';
+
+  @property({ type: String }) viewMode = 'open'; // open | closed
+
+  @property({ type: Number }) hideBannerDays = 7;
+
+  @property({ type: String }) directory = '';
+
+  updated(changed: any) {
+    if (changed.has('iaFiles') || changed.has('iaMD')) {
+      this.shuffledMoments = [];
+      const { directory } = this.iaMD;
+      this.directory = directory;
+    }
+    if (changed.has('directory')) {
+      this.shuffleMoments(this.directory);
+    }
+  }
+
+  shuffleMoments(directory: string) {
+    const moments: TimelineMoment[] = formatMoments(
+      this.iaFiles,
+      this.iaMD,
+      this.landingURL,
+      directory
+    );
+
+    const newMoments = [...moments];
+    this.shuffledMoments = newMoments;
+
+    this.dispatchEvent(
+      new CustomEvent('momentsShuffled', {
+        detail: { moments: this.shuffledMoments },
+      })
+    );
+  }
+
+  private closeBanner() {
+    this.viewMode = 'closed';
+
+    if (this.hideBannerDays > 0) {
+      (window as any)?.Cookies.set('anniv-banner', 'x', {
+        path: '/',
+        expires: this.hideBannerDays,
+        domain: '.archive.org',
+      });
+    }
+  }
+
+  get closeButton() {
+    return html`<button
+      class="close-banner"
+      title="close banner for the day"
+      @click=${() => this.closeBanner()}
+    >
+      ${CloseCircleIcon}
+    </button>`;
+  }
+
+  render() {
+    if (this.viewMode === 'closed') {
+      return nothing;
+    }
+
+    return html`
+      <section class=${this.viewMode}>
+        <a
+          class="left-anchor"
+          href=${this.landingURL}
+          target="_blank"
+          title="Celebrate Internet Archive's 25th Anniversary"
+          rel="nofollow"
+        >
+          ${wayforwardSvgMin} ${wayforwardSvg}
+        </a>
+        <mar-quee
+          class="marquee"
+          .list=${this.shuffledMoments}
+          .defaultLink=${this.landingURL}
+        ></mar-quee>
+        ${this.closeButton}
+      </section>
+    `;
+  }
+
+  static get styles() {
+    const bannerHeight = css`var(--bannerHeight, 50px)`;
+    const marqueeWidth = css`var(--marquee-width, 150%)`;
+    const marqueeAnimation = css`var(--marquee-animation-s, 50s)`;
+
+    return css`
+      :host {
+        --bannerHeight: ${bannerHeight};
+        --marquee-height: ${bannerHeight};
+        --marquee-width: ${marqueeWidth};
+        --marquee-animation-s: ${marqueeAnimation};
+      }
+
+      section,
+      section * > * {
```

Comment:
What is the intended target of this `section * > *` selector? Can we make it more specific so it's more obvious what the target is?

Reply:
mmm how so? this means all the direct children except the last.  are you suggesting I put a class and point to that?

Reply:
Yeah I see you're targeting all of the children and the children's children, but that is casting a wide net that can accidentally capture unintended DOM elements so I was just looking for something more specific.

Reply:
yah, I can scope further 🔭